dataset/src/main/scala/frameless/ops/AggregateTypes.scala

```diff
@@ -0,0 +1,19 @@
+package frameless
+package ops
+
+import shapeless._
+
+trait AggregateTypes[V, U <: HList] {
```

Comment:
Didn't know about `Comapped`. I think it would be better with kind projector, 
```scala
   def agg[TC <: HList, C <: HList, Out0 <: HList, Out1](columns: TC)(
      implicit
      mapped: Comapped.Aux[TC, Lambda[X => TypedAggregate[T, X]], C],
      encoder: TypedEncoder[Out1],
      append: Prepend.Aux[K, C, Out0],
      toTuple: Tupler.Aux[Out0, Out1],
     columnsToList: ToTraversable.Aux[TC, List, UntypedExpression[T]]
   ): TypedDataset[Out1]
```

WDYT?